metriq_gym/run.py

```diff
@@ -112,11 +117,61 @@ def poll_job(args: argparse.Namespace, job_manager: JobManager) -> None:
+        print(results)
+        if is_upload:
+            upload_job(args, job_type, job_data, results, platforms[metriq_job.device_name.lower()], metriq_job.dispatch_time)
     else:
         logger.info("Job is not yet completed. Please try again later.")
 
 
+def upload_job(args: argparse.Namespace, job_type: JobType, job_data: BenchmarkData, result_data: BenchmarkResult, platform: int, dispatch_time: datetime):
+    # Ignored attributes are defined in subclasses, not BenchmarkData and BenchmarkResult
+    client = MetriqClient(os.environ.get("METRIQ_CLIENT_API_KEY"))
+    task = 0
+    method = 0
+    if job_type == JobType.QUANTUM_VOLUME:
+        task = 235 #Quantum Volume task ID
+        method = 144 #Heavy output generation task ID
+    elif job_type == JobType.BSEQ:
+        task = 236 #BSEQ task ID
+        method = 426 #BSEQ method ID
+    else:
+        raise Exception("You're trying to upload an unrecognized job type!")
+    client.submission_add_task(args.submission_id, task)
+    client.submission_add_method(args.submission_id, method)
+    client.submission_add_platform(args.submission_id, platform)
+    result_create_request = ResultCreateRequest(
+        task = str(task),
+        method = str(method),
+        platform = str(platform),
+        isHigherBetter = str(True),
+        metricName = "",
+        metricValue = str(0),
+        evaluatedAt = dispatch_time.strftime("%Y-%m-%d"),
+        qubitCount = str(job_data.num_qubits),                                         # type: ignore[attr-defined]
+        shots = str(job_data.shots),                                                   # type: ignore[attr-defined]
+        # circuitDepth = str | None = None,
+        # notes: str | None = None
+        # sampleSize: str | None = None
+        # standardError: str | None = None
+    )
+    if job_type == JobType.QUANTUM_VOLUME:
+        result_create_request.metricName = "Heavy-output generation rate"
+        result_create_request.metricValue = str(result_data.hog_prob)                  # type: ignore[attr-defined]
+        client.result_add(result_create_request, args.submission_id)
+        result_create_request.metricName = "Cross-entropy benchmark fidelity"
+        result_create_request.metricValue = str(result_data.xeb)                       # type: ignore[attr-defined]
+        client.result_add(result_create_request, args.submission_id)
+    elif job_type == JobType.BSEQ:
+        result_create_request.metricName = "Largest connected component size"
+        result_create_request.metricValue = str(result_data.largest_connected_size)    # type: ignore[attr-defined]
+        client.result_add(result_create_request, args.submission_id)
+        result_create_request.metricName = "Largest connected component size per node"
+        result_create_request.metricValue = str(result_data.fraction_connected)        # type: ignore[attr-defined]
+        client.result_add(result_create_request, args.submission_id)
```

Comment:
This _switch_ in run.py doesn't look scalable. Assigning name/value metriq.info metrics from the benchmark result could be a responsibility of the benchmark. We could have an upload object along the BenchmarkResult object (e.g. https://github.com/unitaryfund/metriq-gym/blob/97b32fd498853354e08047faa3effed298632f14/metriq_gym/benchmarks/qml_kernel.py#L21) which provide the mapping of result fields and metriq.info field.

Reply:
Done.